### PR TITLE
add QA module

### PR DESCRIPTION
**Pull request by andreaturli**
## Continuos Deployment Test

**tl;dr**: this PR adds the QA module to ensure that for each build the SeaClouds platform also gets deployed using Apache Brooklyn onto 2 BYONs

## Automatic deployment
This PR adds to the build pipeline an additional maven plugin called `maven-brooklyn-plugin` that deploys the SeaClouds platform blueprint onto Docker containers.

An `integration-test` goal will be triggered at the end of `install` and will run a brooklyn container and a couple of ubuntu containers that will be the BYON location to deploy the SeaClouds platform. The [maven-brooklyn-plugin](https://github.com/brooklyncentral/brooklyn-maven-plugin) will use the Brooklyn container to deploy the SeaClouds blueprint on the specified location and will check that the main application and all of its entities will be in `service.isUp=true` status.

Notice I have to use Travis's [standard-infrastucture](http://docs.travis-ci.com/user/workers/standard-infrastructure/) instead of the previously used the [container-based-infrastructure](http://docs.travis-ci.com/user/workers/container-based-infrastructure/) so I had to manually implement a caching mechanism based on AWS S3. 

PROS:
- each build will not only compile and run the unit tests and the integration tests (where available) but it will also run a deployment test
- the deployment infrastructure is completely free.

CONS:
- the build time is now ~11 minutes instead of 4-5 minutes as before.

The PR modifies the following:
``` 
modify .travis.yml
add -Ptravis profile
fix cache dirs
use Dockerfile
```

### CODE REVIEW

**Comment by adriannieto** [comment]
This is a great idea, thanks you @andreaturli

**Comment by rosogon** [comment]
Thanks @andreaturli. I haven't tested it, but It's +1 for me.

**Comment by andreaturli** [comment]
https://travis-ci.org/SeaCloudsEU/SeaCloudsPlatform/builds/90762935#L3222-L3227
